File: alchemi-devel/src/Alchemi.Core/Owner/FileDependencyCollection.cs
#region Alchemi copyright and license notice

/*
* Alchemi [.NET Grid Computing Framework]
* http://www.alchemi.net
*
* Title			:	FileDependencyCollection.cs
* Project		:	Alchemi Core
* Created on	:	2003
* Copyright		:	Copyright © 2005 The University of Melbourne
*					This technology has been developed with the support of 
*					the Australian Research Council and the University of Melbourne
*					research grants as part of the Gridbus Project
*					within GRIDS Laboratory at the University of Melbourne, Australia.
* Author         :  Akshay Luther (akshayl@cs.mu.oz.au) and Rajkumar Buyya (raj@cs.mu.oz.au)
* License        :  GPL
*					This program is free software; you can redistribute it and/or 
*					modify it under the terms of the GNU General Public
*					License as published by the Free Software Foundation;
*					See the GNU General Public License 
*					(http://www.gnu.org/copyleft/gpl.html) for more details.
*
*/ 
#endregion

using System;
using System.Collections;
using System.Collections.Specialized;
using Alchemi.Core.Owner;

namespace Alchemi.Core.Owner
{
	/// <summary>
	/// Represents a collection of FileDependencies
	/// </summary>
    [Serializable]
    public class FileDependencyCollection : System.Collections.ReadOnlyCollectionBase
    {
		/// <summary>
		/// Gets the FileDependency with the given index
		/// </summary>
        public FileDependency this[int index]
        {
            get 
            { 
                return (FileDependency) InnerList[index]; 
            }
        }
    
        //-----------------------------------------------------------------------------------------------    
    
		/// <summary>
		/// Adds the given FileDependency object to this collection
		/// </summary>
		/// <param name="dependency">file dependency to add</param>
        public void Add(FileDependency dependency)
        {
            foreach (FileDependency fd in InnerList)
            {
                if (fd.FileName == dependency.FileName)
                {
                    throw new InvalidOperationException("A file dependency with the same name already exists.", null);
                }
            }
            InnerList.Add(dependency);
        }
    }
}

File: alchemiJavaAPI-devel/src/org/gridbus/alchemi/client/FileDependencyCollection.java
/*
 * Title        :  FileDependencyCollection.java
 * Package      :  org.gridbus.alchemi.client
 * Project      :  AlchemiJavaAPI
 * Description	:  
 * Created on   :  4/08/2005
 * Author		:  Krishna Nadiminti (kna@cs.mu.oz.au)
 * Copyright    :  (c) 2005, Grid Computing and Distributed Systems Laboratory, 
 * 					   Dept. of Computer Science and Software Engineering,
 * 					   University of Melbourne, Australia.
 * 
 * This program is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation; either version 2 of the License, or (at your option) any later  version.
 * See the GNU General Public License (http://www.gnu.org/copyleft/gpl.html)for more details.
 * 
 */

package org.gridbus.alchemi.client;

import java.util.ArrayList;

/**
 * @author krishna
 *
 */
public class FileDependencyCollection{
	
	private ArrayList files;
	
	public FileDependencyCollection(){
		files = new ArrayList();
	}
	
	/**
	 * 
	 */
	public int size() {	
		return files.size();
	}

	/**
	 * 
	 */
	public boolean isEmpty() {		
		return files.isEmpty();
	}

	/**
	 * 
	 */
	public boolean contains(FileDependency fileDep) {
		return files.contains(fileDep);
	}

	/**
	 *
	 */
	public FileDependency[] toArray() {
		Object[] obj = files.toArray();
		FileDependency[] fileDeps = new FileDependency[obj.length];
		for (int i=0; i < obj.length; i++)
			fileDeps[i] = (FileDependency)obj[i];
		return fileDeps;
	}

	/**
	 * 
	 */
	public boolean add(FileDependency fileDep) {
		if (this.contains(fileDep))
			return false;
		else
			return files.add(fileDep);
	}

	/**
	 * 
	 */
	public boolean remove(FileDependency fileDep) {
		return files.remove(fileDep);
	}

	/**
	 * 
	 */
	public void clear() {
		files.clear();
	}

	/**
	 * 
	 */
	public FileDependency get(int index) {
		return (FileDependency)files.get(index);
	}

	/**
	 * 
	 */
	public FileDependency set(int index, FileDependency fileDep) {
		return (FileDependency)files.set(index, fileDep);
	}

	/**
	 * 
	 */
	public FileDependency remove(int index) {
		return (FileDependency)files.remove(index);
	}
}
